psycop/projects/scz_bp/feature_generation/outcome_specification/first_scz_or_bp_diagnosis.py

```diff
@@ -0,0 +1,66 @@
+import pandas as pd
+import polars as pl
+
+from psycop.projects.scz_bp.feature_generation.eligible_prediction_times.scz_bp_eligible_config import (
+    N_DAYS_WASHIN,
+)
+from psycop.projects.scz_bp.feature_generation.outcome_specification.add_time_from_first_visit import (
+    add_time_from_first_visit,
+)
+from psycop.projects.scz_bp.feature_generation.outcome_specification.bp_diagnoses import (
+    get_first_bp_diagnosis,
+)
+from psycop.projects.scz_bp.feature_generation.outcome_specification.scz_diagnoses import (
+    get_first_scz_diagnosis,
+)
+
+
+def get_first_scz_or_bp_diagnosis() -> pl.DataFrame:
+    dfs = {"scz": get_first_scz_diagnosis(), "bp": get_first_bp_diagnosis()}
+    for df_type in dfs:
+        dfs[df_type]["source"] = df_type
+
+    combined = pd.concat(
+        [dfs[k] for k in dfs],
+        axis=0,
+    )
+
+    first_scz_or_bp_indicator = (
+        combined.sort_values("timestamp")
+        .groupby("dw_ek_borger")
+        .first()
+        .reset_index(drop=False)
+    )
+    # subtract 1 day from first diagnosis to avoid any leakage and add indicator column
+    first_scz_or_bp_indicator = pl.from_pandas(first_scz_or_bp_indicator).with_columns(
+        pl.col("timestamp") - pl.duration(days=1),
```

Comment:
I assume this overrides the "timestamp" column, tested?

Reply:
Yep!